ReSharper.FSharp/src/FSharp.Psi.Intentions/src/QuickFixes/UpdateRecordFieldsInSignatureFix.fs

```diff
@@ -0,0 +1,85 @@
+open JetBrains.ReSharper.Psi.Tree
+open JetBrains.ReSharper.Resources.Shell
+
+type UpdateRecordFieldsInSignatureFix(error: DefinitionsInSigAndImplNotCompatibleFieldWasPresentError) =
+    inherit FSharpQuickFixBase()
+
+    let getRecordRepresentation (typeDecl: IFSharpTypeDeclaration) =
+        match typeDecl.TypeRepresentation with
+        | :? IRecordRepresentation as rr -> Some rr
+        | _ -> None
+
+    let implementationRecordRepr =
+        let typeDecl = FSharpTypeDeclarationNavigator.GetByIdentifier(error.TypeName)
+        getRecordRepresentation typeDecl
+
+    override this.ExecutePsiTransaction _ =
+        use writeCookie = WriteLockCookie.Create(error.TypeName.IsPhysical())
+        use disableFormatter = new DisableCodeFormatter()
+
+        match implementationRecordRepr with
+        | None -> ()
+        | Some implementationRecordRepr ->
+            let decl = implementationRecordRepr.TypeDeclaration.DeclaredElement
+            let declarations = if isNull decl then Seq.empty else decl.GetDeclarations()
+
+            let signatureRecordRepr =
+                declarations
+                |> Seq.tryPick (fun d ->
+                    match d with
+                    | :? IFSharpTypeDeclaration as signatureTypeDecl when
+                        signatureTypeDecl.GetSourceFile().IsFSharpSignatureFile ->
+                        getRecordRepresentation signatureTypeDecl
+                    | _ -> None)
+
+            match signatureRecordRepr with
+            | None -> ()
+            | Some signatureRecordRepr ->
+                let signatureFieldCount = signatureRecordRepr.FieldDeclarations.Count
+
+                implementationRecordRepr.FieldDeclarations
+                |> Seq.iter (fun implFieldDecl ->
+                    let index = implementationRecordRepr.FieldDeclarations.IndexOf(implFieldDecl)
+                    if index < signatureFieldCount then
+                        // The signature record definition has a field at the current index
+                        // The name or type might be wrong
+                        let signatureFieldDecl = signatureRecordRepr.FieldDeclarations[index]
+                        if implFieldDecl.SourceName = signatureFieldDecl.SourceName &&
+                           implFieldDecl.DeclaredElement.ContainingType = signatureFieldDecl.DeclaredElement.ContainingType then
+                            // fields are identical
+                            ()
+                        elif implFieldDecl.SourceName <> signatureFieldDecl.SourceName then
+                            // field names are different, update signature field name
+                            signatureFieldDecl.SetName(implFieldDecl.NameIdentifier.Name)
+                        else
+                             // Update entire field, I'm not sure how to update the type only.
+                             ModificationUtil.ReplaceChild(signatureFieldDecl :> IFSharpTreeNode, implFieldDecl)
```

Comment:
Using `ModificationUtil.ReplaceChild(signatureFieldDecl.TypeUsage, implFieldDecl.TypeUsage)` also works. 
Is there any reason you would prefer to create an entirely new node instead of replacing it with what we found in the implementation?

I think I understand how to create a new node, just curious about why not reuse it instead.

Reply:
> Is there any reason you would prefer to create an entirely new node instead of replacing it with what we found in the implementation?

Yes, the `open`s might be different, and it might easier to create a type usage in the context of a signature than to update the opens correctly, at least for now.

Reply:
Oh, here's another interesting case... It'll either will work automatically (due to `'A` and `'B` considered equal) or may be quite tricky one to solve otherwise.

```fsharp
// impl
type R<'A> =
    { Field1: int
      Field2: 'A }
```

```fsharp
// sig
type R<'B> =
    { Field1: int
      Field2: 'B }
```

Reply:
> I think I understand how to create a new node, just curious about why not reuse it instead.

I'll likely want to swap it later with a more complicated logic that will be able to both add `open`s and print types correctly, and until that printing types in existing `open`s context seems to be a good compromise between being easy to implement and correct.

Reply:
I think this is better now, let me know if it still needs a tweak.